Standards/scs-0402-v1-status-page-openapi-spec-decision.md

```diff
@@ -82,6 +82,46 @@ An impact defines the relation between an incident and a component. A component
 
 To reflect this, each component and incident can have a list of impacts, stating the type of impact and a reference to the incident or component, it refers to.
 
+Furthermore a severity value is added to the impact when referencing an component, to gauge the impacts severity on the system.
+
+### Severity
+
+A severity contains a name, that MUST be unique and WILL be used as identifier. The value marks the upper boundary of the severity.
+
+The severity value range is calculated by the previous severitys (SeverityA) value + 1 to the current severitys (SeverityB) value. These limits are inclusive.
+
+`SeverityA.value + 1 <=Severity <=SeverityB.value`
+
+Example:
+
+```json
+[
+  {
+    "displayName": "Operational",
+    "value": 10
+  },
+  {
+    "displayName": "Restricted",
+    "value": 50
+  },
+  {
+    "displayName": "Broken",
+    "value": 100
+  }
+]
+```
+
+This means:
+
+- Operational from 0 to 10
```

Comment:
Does that mean that there are incidents affecting services in such little ways that they should not be noticed at all by users?
Like "We're doing maintenance, but you - dear customer - should not be impacted at all"?

Reply:
AFAIK is an "maintenance mode" planned by @snkaupe

Reply:
Indeed. I considered just using incidents with a special severity value (such as 0 or 100) to signify a maintenance event. Thinking about it now, it might be sensible to include this in the specification, so that we can depend upon it in the frontend an backend to detect maintenance incidents. There are a few open questions regarding maintenance events anyway, so we might leave that open right now and return to it at the appropriate time, i.e. when we are discussing maintenance events in depth.

Reply:
Thinking about this, the statement "... should not be impacted at all..." is not quite right either.

Maintenance can include disruptions of the normal usage, but these aren't on accident, like a normal incident, but are well planned.

Reply:
Good argument. Is a maintenance session that is taking down a component not basically the same as any other kind of incident that disrupts said component to a point where it becomes unavailable, just that we are aware of it before it happens? Maybe a maintenance event should either be indicated by a special flag inside an incident or be represented by an entirely different resource.

Reply:
As mentioned by you in https://github.com/SovereignCloudStack/issues/issues/366#issuecomment-1959180141 a  maintenance is a kind of impact, that is known beforehand and currently the API has no way of managing this kind of behavior.

Either we allow for future impacts and handle their retrieval on the API like some kind of special case, or include a new resource for maintenance. I would prefer the later, to keep it more separated from the "normal" impacts. This may be clarified with @fkr today, too.

Reply:
As discussed in the sprint review: A maintenance mode is out of scope for the MVP. So no further considerations about "future issues" at the moment.

Reply:
To address @joshmue original question, an impact might be so insignificant that operation is basically not impacted e.g. Main router is down, but fail over activated a secondary one. An incident, but everything is still working fine.

Its an example after all, the definition of severities is part of the configuration done by the user who wants to run the status-page. What names and values they use is their choice.

Is this discussion resolved with it, or any change requests?

Reply:
Yes, having "Operational" as example severity seems LGTM in that regard.

Resolving this discussion...